frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/moveable-feature-manager.ts

```diff
@@ -0,0 +1,126 @@
+import type { FeatureManager } from '../utility/feature-manager';
+import { MovementAnimator } from '../utility/movement-animator';
+import type {
+    GeometryHelper,
+    GeometryWithCoordinates,
+    PositionableElement,
+    Positions,
+} from '../utility/geometry-helper';
+import type { OpenPopupOptions } from '../utility/popup-manager';
+import { TranslateInteraction } from '../utility/translate-interaction';
+import { ElementManager } from './element-manager';
+
+/**
+ * The base class for all element feature managers.
+ * * manages the position of the element
+ * * manages the default interactions of the element
+ */
+export abstract class MoveableFeatureManager<
+        Element extends PositionableElement,
+        FeatureType extends GeometryWithCoordinates = Point
+    >
+    extends ElementManager<Element, FeatureType, ReadonlySet<keyof Element>>
+    implements FeatureManager<FeatureType>
+{
+    abstract override readonly type: keyof ExerciseState;
+    public readonly togglePopup$ = new Subject<OpenPopupOptions<any>>();
+    protected readonly movementAnimator: MovementAnimator<FeatureType>;
+    constructor(
+        protected readonly olMap: OlMap,
+        public readonly layer: VectorLayer<VectorSource<FeatureType>>,
+        private readonly proposeMovementAction: (
+            newPosition: Positions<FeatureType>,
+            element: Element
+        ) => void,
+        protected readonly geometryHelper: GeometryHelper<FeatureType, Element>
+    ) {
+        super();
+        this.movementAnimator = new MovementAnimator<FeatureType>(
+            this.olMap,
+            this.layer,
+            this.geometryHelper.getNextPosition,
+            this.geometryHelper.getFeatureCoordinates
+        );
+    }
+
+    override unsupportedChangeProperties: ReadonlySet<keyof Element> = new Set(
+        [] as const
+    );
+    createFeature(element: Element): Feature<FeatureType> {
+        const elementFeature = this.geometryHelper.create(element);
+        elementFeature.setId(element.id);
+        this.layer.getSource()!.addFeature(elementFeature);
+        TranslateInteraction.onTranslateEnd<FeatureType>(
+            elementFeature,
+            (newPosition) => {
+                this.proposeMovementAction(newPosition, element);
+            },
+            this.geometryHelper.getFeaturePosition
+        );
+        return elementFeature;
+    }
+
+    isFeatureTranslatable(feature: Feature<FeatureType>) {
+        return true;
+    }
+
+    deleteFeature(
+        element: Element,
+        elementFeature: Feature<FeatureType>
+    ): void {
+        this.layer.getSource()!.removeFeature(elementFeature);
+        elementFeature.dispose();
+        this.movementAnimator.stopMovementAnimation(elementFeature);
+    }
+
+    changeFeature(
+        oldElement: Element,
+        newElement: Element,
+        // It is too much work to correctly type this param with {@link unsupportedChangeProperties}
```

Comment:
You cannot type this param with `unsupportedChangeProperties` because that is also just `ReadonlySet<keyof Element>`

Reply:
We just leave it as it is.